modules/packages/LinearAlgebra.chpl

```diff
@@ -841,4 +843,467 @@ proc trace(A: [?D] ?eltType) {
     return trace;
 }
 
+//
+// Type helpers
+//
+private proc isDefaultArr (A: []) param where A.domain._value.dist: DefaultDist { return true; }
+private proc isDefaultArr (A: []) param { return false; }
+
+private proc isDefaultDom (D: domain) param where D._value.dist: DefaultDist { return true; }
+private proc isDefaultDom (D: domain) param { return false; }
+
+
+/* Linear Algebra Sparse Submodule
+
+A high-level interface to linear algebra operations and procedures for sparse
+matrices (2D arrays).
+
+Sparse matrices are represented as 2D arrays domain-mapped to a sparse *layout*.
+Only the ``CS(compressRows=true)`` (CSR) layout of the
+`:mod:`LayoutCS` layout module is currently supported.
+
+See the `Sparse Primer <primers-sparse>`_ for more information about working
+with sparse domains and arrays in Chapel.
+
+Sparse Linear Algebra Interface
+-------------------------------
+
+``LinearAlgebra.Sparse`` follows the same conventions and interface choices
+ as the parent module, ``LinearAlgebra``, with few exceptions. These
+ exceptions are detailed below.
+
+
+**Sparse Domains**
+
+In Chapel, implicit changes to the index set of a sparse array must be made
```

Comment:
I don't think the word 'implicit' belongs here. Changes made directly to the domain would not be implicit.